cmd/gardener-extension-provider-kubevirt/app/app.go

```diff
@@ -0,0 +1,181 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+ */
+
+package app
+
+import (
+	"context"
+	"fmt"
+	apiskubevirthelper "github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt/helper"
+	webhookcmd "github.com/gardener/gardener/extensions/pkg/webhook/cmd"
+	"os"
+
+	kubevirtinstall "github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt/install"
+	kubevirtcmd "github.com/gardener/gardener-extension-provider-kubevirt/pkg/cmd"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/controller/healthcheck"
+	kubevirtinfra "github.com/gardener/gardener-extension-provider-kubevirt/pkg/controller/infrastructure"
+	kubevirtworker "github.com/gardener/gardener-extension-provider-kubevirt/pkg/controller/worker"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/kubevirt"
+	kubevirtcontrolplane "github.com/gardener/gardener-extension-provider-kubevirt/pkg/webhook/controlplane"
+	kubevirtcontrolplaneexposure "github.com/gardener/gardener-extension-provider-kubevirt/pkg/webhook/controlplaneexposure"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	controllercmd "github.com/gardener/gardener/extensions/pkg/controller/cmd"
+	"github.com/gardener/gardener/extensions/pkg/controller/worker"
+	"github.com/gardener/gardener/extensions/pkg/util"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
+	var (
+		restOpts = &controllercmd.RESTOptions{}
+		mgrOpts  = &controllercmd.ManagerOptions{
+			LeaderElection:          true,
+			LeaderElectionID:        controllercmd.LeaderElectionNameID(kubevirt.Name),
+			LeaderElectionNamespace: os.Getenv("LEADER_ELECTION_NAMESPACE"),
+			WebhookServerPort:       443,
+		}
+		configFileOpts = &kubevirtcmd.ConfigOptions{}
+
+		// options for the infrastructure controller
+		infraCtrlOpts = &controllercmd.ControllerOptions{
+			MaxConcurrentReconciles: 5,
+		}
+		reconcileOpts = &controllercmd.ReconcilerOptions{}
+
+		// options for the health care controller
+		healthCareCtrlOpts = &controllercmd.ControllerOptions{
```

Comment:
Please rename this to `healthCheckCtrlOpts`, since it's "healthcheck controller", not "healthcare controller"

Reply:
done